File: presentation/src/main/java/com/mbmc/template/ui/UiSate.java
package com.mbmc.template.ui;

import androidx.databinding.ObservableBoolean;

public class UiSate {

    public final ObservableBoolean hasText = new ObservableBoolean();
    public final ObservableBoolean hasError = new ObservableBoolean();
    public final ObservableBoolean isLoading = new ObservableBoolean();

    public void setHasText(boolean hasText) {
        this.hasText.set(hasText);
    }

    public void setHasError() {
        hasError.set(true);
    }

    public void setIsLoading() {
        isLoading.set(true);
    }

    public void setIsIdle() {
        hasError.set(false);
        isLoading.set(false);
    }

}

File: feature/search/src/main/kotlin/com/mbmc/template/feature/search/presentation/ui/UiSate.kt
package com.mbmc.template.feature.search.presentation.ui

import androidx.databinding.ObservableBoolean

class UiSate {
    val hasText = ObservableBoolean()
    val hasError = ObservableBoolean()
    val isLoading = ObservableBoolean()

    fun setHasText(hasText: Boolean) {
        this.hasText.set(hasText)
    }

    fun setHasError() {
        hasError.set(true)
    }

    fun setIsLoading() {
        isLoading.set(true)
    }

    fun setIsIdle() {
        hasError.set(false)
        isLoading.set(false)
    }
}
